worker/worker_test.go

```diff
@@ -49,5 +50,28 @@ func TestWorkerStartsConsequently(t *testing.T) {
 	cancel()
 	wg.Wait()
 
-	assert.Equal(t, 3, counter, "Should execute 3 times - 1st immidietly, and 2 after with delay between runs")
+	assert.Equal(t, 3, counter, "Should execute 3 times - 1st immediately, and 2 after with delay between runs")
+}
+
+func TestWorkerStartsWithoutExecution(t *testing.T) {
+	counter := 0
+	options := worker.DefaultWorkerOptions(100 * time.Millisecond)
+	options.Interval = -1
+
+	worker := worker.NewWorkerBuilder("test", func() error {
+		time.Sleep(100 * time.Millisecond)
+		counter++
+		return nil
+	}).WithOptions(options).Build()
+
+	wg := &sync.WaitGroup{}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	worker.Start(ctx, wg)
+
+	time.Sleep(350 * time.Millisecond)
```

Comment:
I am a bit concerned with the sleep, 350 IMO is too much. We should try to minimize the sleep. How about doing it this way?

- We remove time.Sleep() at line 62
- replace line 70-76 with this: 
```
worker.Start(ctx, wg)
go func() {
  <-time.After(50 * time.Millisecond)
  cancel()
}()
wg.Wait()

assert.Equal(t, 0, counter)
```

Since RunImmediately is true, we know 100% after wg.Wait(), there will be at least 1 execution of the worker, so I think this test is valid good.

Reply:
oh, we can use ctx timeout too

```
wg := &sync.WaitGroup{}
ctx, cancel := context.WithTimeout(context.Background(), 50 * time.Millisecond)
defer cancel()

wg.Wait()
assert.Equal(t, 0, counter)
```

Reply:
```
Since RunImmediately is true, we know 100% after wg.Wait(), there will be at least 1 execution of the worker, whichever it is:

If the hold function is run, the test will immediately terminate
If the start function is run, we will only wait for 50 ms
```
^ this is not true, both will wait till 50ms

need further modification if we want to terminate it immediately when start function is run
when the function behaves correctly (hold is run), I think we anyway need to wait till 50m, there's no way around it